enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1148 @@
+    ControlPlaneExternalLBSubnetRole SubnetRoleType = "ControlPlaneExternalLB" 
+
+    // ControlPlaneInternalLBSubnetRole specifies subnets used by the internal control plane
+    // load balancer that serves the Kubernetes API server.
+    ControlPlaneInternalLBSubnetRole SubnetRoleType = "ControlPlaneInternalLB"
+)
+```
+
+The `platform.aws.vpc.subnets` field will be introduced under the TechPreview feature gate, and `platform.aws.subnets`
+will be deprecated when `platform.aws.vpc.subnets` graduates to GA.
+
+##### Installer Validation Rules
+
+These validation rules only apply to the new `platform.aws.vpc.subnets` field. These validations aim to improve
+user experience, but are not permanent and may be updated or removed to accommodate future use cases.
+
+###### All or Nothing Subnet Roles Selection
+
+This proposed API follows an "all or nothing" approach for roles: either all required roles (see
+[Automatic vs. Manual Role Selection](#automatic-vs-manual-role-selection) for list) must be explicitly assigned
+(manual role selection), or no roles must be assigned (automatic role selection). The installer must not allow a mix of
+automatic and manual role assignments, as this could lead to confusing behavior. See [Allowing Mixed Automatic and Manual Subnet Selection Alternative](#allowing-mixed-automatic-and-manual-subnet-selection-alternative) for the reason behind this validation.
+
+###### All Subnets Belong to the Same VPC Validation
+
+The [existing validation](https://github.com/openshift/installer/blob/0d77aa8df5ddc68e926aa11da24a87981021b256/pkg/asset/installconfig/aws/subnet.go#L91)
+for the deprecated `platform.aws.subnets` field ensuring all subnets belong to the same VPC
+should apply to all subnets specified in `platform.aws.vpc.subnets`.
+
+###### Multiple Load Balancer Subnets in the Same AZ Validation
+
+The installer must reject multiple IngressControllerLB subnets in the same AZ as this will be
+rejected by the AWS CCM.
+
+###### The Old and New Subnets Fields Cannot be Specified Together Validation
+
+Since `platform.aws.subnets` will be deprecated, the installer must not allow `platform.aws.vpc.subnets` and
+`platform.aws.subnets` to both be specified.
+
+###### Maximum of 10 IngressController Subnets Validation
+
+Since the IngressController's API [only allows 10 subnets](https://github.com/openshift/api/blob/ee5bb7eaf6b6638d4e3b33ba4ff0834212cdb75d/operator/v1/types_ingress.go#L564),
+the installer must not allow more than 10 IngressController subnets as well.
+
+###### Consistent Cluster Scope with IngressControllerLB Subnets Validation
+
+The installer must not allow **any** public IngressControllerLB subnets for private clusters (`installconfig.publish: Internal`)
+as this is a security risk as discussed in the [Do Not Use Untagged Subnets User Stories](#do-not-use-untagged-subnets-user-stories) section.
+
+Conversely, the installer must now allow **any** private IngressControllerLB subnets for public clusters
```

Comment:
@gcs278 Adding this for future reference and removing the hold to signal my review.  If I'm wrong about this, please let me know.
```suggestion
Conversely, the installer must not allow **any** private IngressControllerLB subnets for public clusters
```

Reply:
Whoops, yea that was a typo, good catch. If I have time tomorrow  I might follow up with a tiny pr to fix this.